### PR TITLE
Formatters and validators

### PR DESCRIPTION
**Pull request by martypowell**
added initial revisions to the project.

1. added to the utility namespace
2. tests contain two files, the spec.js file and the spec.html files. This was used to get around using the namespace.

TODO: fix the two number formats that aren't working (not comment), and add tests where needed.

### CODE REVIEW

**Comment by martypowell** [comment]
@danfox01 fyi

File: js/utility/formatters.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+namespacer('baltimoreCounty.utility');
+
+baltimoreCounty.utility.format = (function () {
+    var _formatters = {
+        currency: function formatCurrency(input) {
```
Comment (mxsnyder) [inline js/utility/formatters.js]: Not a big deal, but would you move these functions out of your object declaration? For example:
```
function whatnot() {}

var thing = { 
    stuff: whatnot 
};
```

File: js/utility/formatters.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            else {
+                console.error("Incorrect Format. Double Check your values.");
+                return null;
+            }
+
+            return format;
+        }
+    };
+
+    return function format(key, val, strFormat) {
```
Comment (mxsnyder) [inline js/utility/formatters.js]: Same thing here, move the function out and just return the identifier.

File: js/utility/validators.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+namespacer('baltimoreCounty.utility');
+
+baltimoreCounty.utility.validate = (function () {
```
Comment (mxsnyder) [inline js/utility/validators.js]: Same comment about moving the functions here.

File: spec/js/utility/formatters.spec.html [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    <body>
+      <div id="mocha"></div>
+
+      <script src="https://cdn.rawgit.com/jquery/jquery/2.1.4/dist/jquery.min.js"></script>
+      <script src="https://cdn.rawgit.com/Automattic/expect.js/0.3.1/index.js"></script>
+      <script src="https://cdnjs.cloudflare.com/ajax/libs/sinon.js/3.3.0/sinon.min.js"></script>
+      <script src="https://cdn.rawgit.com/mochajs/mocha/2.2.5/mocha.js"></script>
+
+      <script>mocha.setup('bdd')</script>
+      <script src="../../../js/utility/namespacer.js"></script>
```
Comment (mxsnyder) [inline spec/js/utility/formatters.spec.html]: For the sake of readability, would you break these up into sections for dependencies (namespacer), scripts, and specs? _I_ know which is which, but if I get hit by a truck, which tends to happen around here, whoever's next might not be too sure what's up.

File: spec/js/utility/validators.spec.html [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    <body>
+      <div id="mocha"></div>
+
+      <script src="https://cdn.rawgit.com/jquery/jquery/2.1.4/dist/jquery.min.js"></script>
+      <script src="https://cdn.rawgit.com/Automattic/expect.js/0.3.1/index.js"></script>
+      <script src="https://cdnjs.cloudflare.com/ajax/libs/sinon.js/3.3.0/sinon.min.js"></script>
+      <script src="https://cdn.rawgit.com/mochajs/mocha/2.2.5/mocha.js"></script>
+
+      <script>mocha.setup('bdd')</script>
+      <script src="../../../js/utility/namespacer.js"></script>
```
Comment (mxsnyder) [inline spec/js/utility/validators.spec.html]: Same comment as above.

Comment (martypowell) [reply spec/js/utility/validators.spec.html]: @mxsnyder all items as listed above have been addressed

**Pull request merged**